### PR TITLE
add FallbackResource alternative for assets/bundles in nginx

### PR DESCRIPTION
**Pull request by fliespl**
As described above - this location rule provides similar functionality to apache FallbackResource example.

### CODE REVIEW

**Comment by OskarStark** [comment]
Thank you for your contribution, sadly I never used this before, so we need to find someone who can verify this.

**Review by Pierstoval — APPROVED** [review]
Seems like a nice proposal, I see no "no" for it, and I'm already using it in production 👍

File: setup/web_server_configuration.rst [hunk 1]
```diff
@@ -287,6 +287,13 @@ The **minimum configuration** to get your application running under Nginx is:
             # try to serve file directly, fallback to index.php
             try_files $uri /index.php$is_args$args;
         }
+        
+        # optionally disable falling back to php script for the asset directories
+        # which will allow nginx to return a 404 error when files are
+        # not found instead of passing the request to Symfony
+        location /bundles {
+            try_files $uri $uri/ =404;
```
Comment (lyrixx) [inline setup/web_server_configuration.rst]: Actually, if you are looking for a file, the `$uri/` lookup seems useless to me.

File: setup/web_server_configuration.rst [hunk 2]
```diff
@@ -287,6 +287,13 @@ The **minimum configuration** to get your application running under Nginx is:
             # try to serve file directly, fallback to index.php
             try_files $uri /index.php$is_args$args;
         }
+        
+        # optionally disable falling back to php script for the asset directories
```
Comment (lyrixx) [inline setup/web_server_configuration.rst]:
```suggestion
        # optionally disable falling back to PHP script for the asset directories
```

Comment (Tobion) [reply setup/web_server_configuration.rst]: why does it say "optionally" if it's enabled by default? I also don't really see the value of adding it. It only prevents use-cases like having a route starting with /bundles.

**Comment by ndench** [comment]
I don't see a downside for adding this, though you might want to add more info as to why you would want this and what are the downsides?

ie.
- this prevents any PHP files in the `/bundles` directory from being executed
- you will need to change `/bundles` to be whatever directory your assets are in
- this will mean that users are presented with nginx 404 pages instead of Symfony 404 pages which you may have custom styling for

**Comment by javiereguiluz** [comment]
Thank you all for the review! I've merged but made some changes to comment this by default (as @Tobion asked), remove the unneeded `$uri/` lookup (as @lyrixx said) and mention that you won't see Symfony's 404 error page (as @ndench explained; thanks for listing the disadvantages ... I think that the first two were not that common to mention them).

**Pull request merged**